src/scripts/main.js

```diff
@@ -1,3 +1,10 @@
 'use strict';
 
-// write code here
+const wall = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+
+const xPosition = wall.offsetWidth / 2 - spider.offsetWidth / 2;
+const yPosition = wall.offsetHeight / 2 - spider.offsetHeight / 2;
+
+spider.style.left = `${xPosition}px`;
+spider.style.top = `${yPosition}px`;
```

Comment:
This code sets the position of the spider only once when the script is loaded. If the dimensions of the wall or the spider change afterwards (due to window resizing, dynamic styling, etc.), the spider will not be repositioned. Consider adding an event listener for the 'resize' event to handle this.